File: 02-programacion-basica/04-operadores-y-expresiones/java/ejemplos-guiados/OperadoresYPrecedencia.java
public class OperadoresYPrecedencia {
    public static void main(String[] args) {
        int a = 8;
        int b = 3;

        // La multiplicación tiene mayor precedencia que la suma.
        int porPrecedencia = a + b * 2;
        int conParentesis = (a + b) * 2;

        // Una expresión lógica combina comparaciones y produce boolean.
        boolean condicion = a > b && b != 0;

        System.out.println("Precedencia: " + porPrecedencia);
        System.out.println("Con paréntesis: " + conParentesis);
        System.out.println("Condición: " + condicion);
    }
}

File: 02-programacion-basica/04-operadores-y-expresiones/scala/ejemplos-guiados/OperadoresYPrecedencia.scala
object OperadoresYPrecedencia:
  def main(args: Array[String]): Unit =
    val a = 8
    val b = 3

    val porPrecedencia = a + b * 2
    val conParentesis = (a + b) * 2
    val condicion = a > b && b != 0

    println(s"Precedencia: $porPrecedencia")
    println(s"Con paréntesis: $conParentesis")
    println(s"Condición: $condicion")
